aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/internal/DelegatingEventLoopGroup.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ScheduledFuture;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import software.amazon.awssdk.annotation.SdkTestInternalApi;
+
+/**
+ * {@link EventLoopGroup} that just delegates to another {@link EventLoopGroup}. Useful for extending or building decorators.
+ */
+public abstract class DelegatingEventLoopGroup implements EventLoopGroup {
+
+    private final EventLoopGroup delegate;
+
+    public DelegatingEventLoopGroup(EventLoopGroup delegate) {
+        this.delegate = delegate;
+    }
+
+    /**
+     * @return The {@link EventLoopGroup} being delegated to.
+     */
+    @SdkTestInternalApi
+    EventLoopGroup getDelegate() {
```

Comment:
Ohhh I didn't see this, again it just feels dirty to expose 'test only' methods like this - even if they are package private. We shouldn't let this leak in here just because this thing is created from a static method.